core/src/main/scala/cats/instances/seq.scala

```diff
@@ -133,8 +133,11 @@ trait SeqInstances extends cats.kernel.instances.SeqInstances {
       final override def traverse[G[_], A, B](fa: Seq[A])(f: A => G[B])(implicit G: Applicative[G]): G[Seq[B]] =
         G.map(Chain.traverseViaChain(fa.toIndexedSeq)(f))(_.toVector)
 
+      override def mapAccumulate[S, A, B](init: S, fa: Seq[A])(f: (S, A) => (S, B)): (S, Seq[B]) =
+        StaticMethods.mapAccumulateFromStrictFunctor(init, fa, f)(this)
```

Comment:
I don't think is actually safe. `Seq` is an interface and I don't think it makes any promise about the order that `map` is applied. I think something like:
```scala
val bldr: List[B] = List.newBuilder[B]
val it = fa.iterator
var s = init
while (it.hasNext) {
  val a = it.next()
  val (nextS, b) = f(s, a)
  bldr += b
  s = nextS
}
(s, bldr.result())
```

Reply:
for Scala 2.13+ the following might be even better:
```
val bldr = fs.iterableFactory.newBuilder[B]
```
but I don't remember if it works for Scala 2.12.

Reply:
> I don't think it makes any promise about the order that map is applied

AFAIK, the only thing `Seq` promises is the order of the elements.
However, I just realized that it doesn't guarantee that the collection will be strict, mainly because both `Stream` and `LazyList` are `Seqs`

I will just remove this, `Seq` should have never had instances IMHO.